libs/sdk-core/src/persist/send_pays.rs

```diff
@@ -72,22 +72,27 @@ impl SqliteStorage {
         Ok(())
     }
 
-    pub(crate) fn list_send_pays(&self, hashes: &[Vec<u8>]) -> PersistResult<Vec<SendPay>> {
+    pub(crate) fn list_send_pays(
+        &self,
+        hash_groups: &[(Vec<u8>, u64)],
+    ) -> PersistResult<Vec<SendPay>> {
         let conn = self.get_connection()?;
         let mut stmt = conn.prepare(
             r#"SELECT created_index, updated_index, groupid, partid, 
                payment_hash, status, amount_msat, destination, created_at, 
                amount_sent_msat, label, bolt11, description, bolt12, 
                payment_preimage, erroronion
                FROM send_pays
-               WHERE payment_hash = :payment_hash"#,
+               WHERE payment_hash = :payment_hash AND groupid = :groupid
+               ORDER BY created_index"#,
         )?;
         let mut send_pays = Vec::new();
-        for hash in hashes {
+        for hash_group in hash_groups {
```

Comment:
I think we can use IN filter expression to get all these in one shot but this is also good for now. Only the first will probably end with lots of queries.